### PR TITLE
Show autosave configure notifcation after first typing in play mode

### PR DESCRIPTION
**Pull request by krasnotsvetov**

### CODE REVIEW

File: unity/EditorPlugin/AfterUnity56/Navigation/Initialization.cs [hunk 1]
```diff
@@ -57,6 +59,32 @@ public static void OnModelInitializationHandler(UnityModelAndLifetime modelAndLi
...
       });
+      
+      modelValue.ShowPreferences.Advise(connectionLifetime, result =>
+      {
+        if (result != null)
+        {
+          MainThreadDispatcher.Instance.Queue(() =>
+          {
+
+            var tab = UnityUtils.UnityVersion >= new Version(2018, 2) ? "_General" : "Rider";
```
Comment (van800) [inline unity/EditorPlugin/AfterUnity56/Navigation/Initialization.cs]: adding try-catch-log may be good idea here.

File: rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/notifications/UnityAutoSaveConfigureNotification.kt [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                        if (!unityHost.model.scriptCompilationDuringPlay.hasValue)
+                            return
+
+                        if (unityHost.model.scriptCompilationDuringPlay.valueOrThrow != ScriptCompilationDuringPlay.RecompileAndContinuePlaying)
+                            return
+
+                        if (!lifetimeDefinition.isAlive)
+                            return
+
+                        val generalSettings = GeneralSettings.getInstance()
```
Comment (kskrygan) [inline rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/notifications/UnityAutoSaveConfigureNotification.kt]: let's move GeneralSettings.getInstance out of this scope, so it won't be calculated for every doc change

File: rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/notifications/UnityAutoSaveConfigureNotification.kt [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    }
+
+    init {
+        solutionLifecycleHost.isBackendLoaded.whenTrue(componentLifetime) {
+            if (!propertiesComponent.getBoolean(settingName) && unityProjectDiscoverer.isUnityProject) {
+
+                val eventMulticaster = EditorFactory.getInstance().eventMulticaster
+                val documentListener: DocumentListener = object : DocumentListener {
+                    override fun documentChanged(event: DocumentEvent) {
+                        if (unityHost.model.editorState.valueOrDefault(EditorState.Disconnected) != EditorState.ConnectedPlay)
```
Comment (kskrygan) [inline rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/notifications/UnityAutoSaveConfigureNotification.kt]: if this works for o(1) then it's fine.

File: rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/notifications/UnityAutoSaveConfigureNotification.kt [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+                            return
+
+                        if (!unityHost.model.scriptCompilationDuringPlay.hasValue)
+                            return
+
+                        if (unityHost.model.scriptCompilationDuringPlay.valueOrThrow != ScriptCompilationDuringPlay.RecompileAndContinuePlaying)
```
Comment (kskrygan) [inline rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/notifications/UnityAutoSaveConfigureNotification.kt]: why valueOrThrow? Isn't throwing assertions here too aggressive?

File: rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/notifications/UnityAutoSaveConfigureNotification.kt [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+            val panel = EditorNotificationPanel(LightColors.RED)
+            panel.setText("You are modifying a script while Unity Editor is being in Play Mode. This can lead to a loss of the state in your running game.")
+
+            panel.createActionLabel("Configure Unity Editor") {
+                unityHost.model.showPreferences.fire()
+                lifetimeDefinition.terminate()
+            }
+
+            panel.createActionLabel("Don't show again") {
```
Comment (kskrygan) [inline rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/notifications/UnityAutoSaveConfigureNotification.kt]: "Do not show again"
Please avoid using "don't", "ain't" and other reductions in UI

File: rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/notifications/UnityAutoSaveConfigureNotification.kt (hunk 2 above)
Comment (krasnotsvetov) [reply rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/notifications/UnityAutoSaveConfigureNotification.kt]: Ooops. Thanks!

File: rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/notifications/UnityAutoSaveConfigureNotification.kt (hunk 3 above)
Comment (krasnotsvetov) [reply rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/notifications/UnityAutoSaveConfigureNotification.kt]: AFAIK, our model properties cache value

**Pull request merged**